### PR TITLE
feature: adding batch support for narrow_copy operator

### PR DESCRIPTION
**Pull request by srossross**
Implement batch support https://github.com/pytorch/functorch/issues/825 for narrow copy

narrow_copy was already added as an opinfo 

cc @zou3519 @Chillee @samdow @soumith

### CODE REVIEW

File: aten/src/ATen/functorch/BatchRulesViews.cpp [hunk 1]
```diff
@@ -490,6 +490,18 @@ std::tuple<Tensor, optional<int64_t>> unfold_batch_rule(
...
 }
 
+std::tuple<Tensor, optional<int64_t>> narrow_copy_batch_rule(
+    const Tensor &self, optional<int64_t> self_bdim, int64_t dim, c10::SymInt start, c10::SymInt length)
+{
+  TORCH_INTERNAL_ASSERT(self_bdim.has_value());
+  auto self_ = moveBatchDimToFront(self, self_bdim);
+  auto logical_rank = rankWithoutBatchDim(self, self_bdim);
+  dim = maybe_wrap_dim(dim, logical_rank) + 1;
+  auto result = self_.narrow_copy_symint(dim, start, length);
```
Comment (srossross) [inline aten/src/ATen/functorch/BatchRulesViews.cpp:500]: @amjames @kshitij12345 pending tests on CI, I think this looks ok, am I missing any guards or edge cases?

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks @srossross

**Comment by srossross** [comment]
What labels should I add to make that lint pass?

**Review by zou3519 — APPROVED** [review]

**Comment by zou3519** [comment]
I believe it wanted a label that begins with `release note:`

**Pull request closed without merging**